lib/DataDrivenLux/src/algorithms/reinforce.jl

```diff
@@ -0,0 +1,105 @@
+abstract type AbstractRewardScale{risk} end
+
+"""
+$(TYPEDEF)
+
+Scales the losses in such a way that the minimum loss is equal to one.
+"""
+struct RelativeReward{risk} <: AbstractRewardScale{risk} end
+
+RelativeReward(risk_seeking = true) = RelativeReward{risk_seeking}()
+
+function (::RelativeReward)(losses::Vector{T}) where {T <: Number}
+    exp.(minimum(losses) .- losses)
+end
+
+function (::RelativeReward{true})(losses::Vector{T}) where {T <: Number}
+    r = exp.(minimum(losses) .- losses)
+    r .- minimum(r)
+end
+
+"""
+$(TYPEDEF)
+
+Scales the losses in such a way that the minimum loss is the most influencial reward.
+"""
+struct AbsoluteReward{risk} <: AbstractRewardScale{risk} end
+
+AbsoluteReward(risk_seeking = true) = AbsoluteReward{risk_seeking}()
+
+function (::AbsoluteReward)(losses::Vector{T}) where {T <: Number}
+    exp.(-losses)
+end
+
+function (::AbsoluteReward{true})(losses::Vector{T}) where {T <: Number}
+    r = exp.(-losses)
+    r .- minimum(r)
+end
+"""
+$(TYPEDEF)
+
+Uses the REINFORCE algorithm to search over the space of possible solutions to the 
+symbolic regression problem.
+
+# Fields
+$(FIELDS)
+"""
+@with_kw struct Reinforce{F, A, L, O, R} <: AbstractDAGSRAlgorithm
+    "Reward function which should convert the loss to a reward."
+    reward::R = RelativeReward(false)
+    "The number of candidates to track"
+    populationsize::Int = 100
+    "The functions to include in the search"
+    functions::F = (sin, exp, cos, log, +, -, /, *)
+    "The arities of the functions"
+    arities::A = (1, 1, 1, 1, 2, 2, 2, 2)
+    "The number of layers"
+    n_layers::Int = 1
+    "Include skip layers"
+    skip::Bool = true
+    "Evaluation function to sort the samples"
+    loss::L = aicc
+    "The number of candidates to keep in each iteration"
+    keep::Union{Real, Int} = 0.1
+    "Use protected operators"
+    use_protected::Bool = true
+    "Use distributed optimization and resampling"
+    distributed::Bool = false
+    "Use threaded optimization and resampling - not implemented right now."
```

Comment:
would it ever be smart? If you're already using BLAS, you're already multithreading

Reply:
I've had a discussion on that with @mohamed82008 and @korsbo . Seems to work fine.